FBTweak/FBTweak.m

```diff
@@ -175,38 +203,59 @@ - (void)setCurrentValue:(FBTweakValue)currentValue
     }
   }
 
-  if (_currentValue != currentValue) {
-      
-    for (id<FBTweakObserver> observer in [_observers setRepresentation]) {
-      if ([observer respondsToSelector:@selector(tweakWillChange:)]) {
-        [observer tweakWillChange:self];
-      }
-    }
-      
-    _currentValue = currentValue;
-    // we can't store UIColor to the plist file. That is why we archive value to the NSData.
-    [[NSUserDefaults standardUserDefaults] setObject:[NSKeyedArchiver archivedDataWithRootObject:_currentValue] forKey:_identifier];
+  [super setCurrentValue:currentValue];
+}
 
-    for (id<FBTweakObserver> observer in [_observers setRepresentation]) {
-      [observer tweakDidChange:self];
-    }
-  }
+- (void)reset {
+  [super reset];
+  self.currentValue = nil;
 }
 
-- (void)addObserver:(id<FBTweakObserver>)observer
+@end
+
+@implementation FBPersistentTweak
+
+- (instancetype)initWithCoder:(NSCoder *)coder
 {
-  if (_observers == nil) {
-    _observers = [NSHashTable weakObjectsHashTable];
+  NSString *identifier = [coder decodeObjectForKey:@"identifier"];
+  NSString *name = [coder decodeObjectForKey:@"name"];
+  NSString *defaultValue = [coder decodeObjectForKey:@"defaultValue"];
```

Comment:
All the above are `Nullable`, and should be treated accordingly.

Reply:
Removed `NSCoding`